meson.build

```diff
@@ -0,0 +1,362 @@
+cdata.set_quoted('FLATPAK_CONFIGDIR', get_option('prefix') / get_option('sysconfdir') / 'flatpak')
+cdata.set_quoted('FLATPAK_BASEDIR', get_option('prefix') / get_option('datadir') / 'flatpak')
+cdata.set_quoted('FLATPAK_TRIGGERDIR', get_option('prefix') / get_option('datadir') / 'flatpak' / 'triggers')
+cdata.set_quoted('LIBEXECDIR', get_option('prefix') / get_option('libexecdir'))
+cdata.set_quoted('DATADIR', get_option('prefix') / get_option('datadir'))
+cdata.set_quoted('LOCALEDIR', get_option('prefix') / get_option('localedir'))
+cdata.set_quoted('SYSTEM_FONTS_DIR', get_option('system_fonts_dir'))
+cdata.set_quoted('SYSTEM_HELPER_USER', get_option('system_helper_user'))
+cdata.set_quoted('SYSTEM_FONT_CACHE_DIRS', ':'.join(get_option('system_font_cache_dirs')))
+cdata.set_quoted('G_LOG_DOMAIN', 'flatpak')
+cdata.set_quoted('GETTEXT_PACKAGE', 'flatpak')
+
+if get_option('system_bubblewrap') == ''
+  cdata.set_quoted('HELPER', get_option('prefix') / get_option('libexecdir') / 'flatpak-bwrap')
+else
+  cdata.set_quoted('HELPER', get_option('system_bubblewrap'))
+endif
+
+if get_option('system_dbus_proxy') == ''
+  cdata.set_quoted('DBUSPROXY', get_option('prefix') / get_option('libexecdir') / 'flatpak-dbus-proxy')
+else
+  cdata.set_quoted('DBUSPROXY', get_option('system_dbus_proxy'))
+endif
+
+# Flatpak is Linux-specific, so for now we assume that -fvisibility=hidden
+# is always supported
+cdata.set('FLATPAK_EXTERN', '__attribute__((visibility("default"))) extern')
+
+if glib_dep.version().version_compare('>=2.60')
+  # Ignore massive GTimeVal deprecation warnings in 2.62
+  cdata.set('GLIB_VERSION_MIN_REQUIRED', 'GLIB_VERSION_2_60')
+endif
+
+if appstream_dep.version().version_compare('>=0.14.0')
+  cdata.set('HAVE_APPSTREAM_0_14_0', 1)
+endif
+
+if libseccomp_dep.found()
+  cdata.set('ENABLE_SECCOMP', 1)
+endif
+
+if xau_dep.found()
+  cdata.set('ENABLE_XAUTH', 1)
+endif
+
+if not get_option('sandboxed_triggers')
+  cdata.set('DISABLE_SANDBOXED_TRIGGERS', 1)
+endif
+
+if cc.has_function(
+  'archive_read_support_filter_all',
+  dependencies : libarchive_dep,
+  prefix : '#include <archive.h>',
+)
+  cdata.set('HAVE_ARCHIVE_READ_SUPPORT_FILTER_ALL', 1)
+endif
+
+configure_file(
+  output : 'config.h',
+  configuration : cdata,
+)
+
+if meson.version().version_compare('>=0.53.0')
+  summary({
+    'Build system helper' : build_system_helper,
+    'Build selinux module' : build_selinux_module,
+    'Build bubblewrap' : (get_option('system_bubblewrap') == ''),
+    'Build dbus-proxy' : (get_option('system_dbus_proxy') == ''),
+    'Use sandboxed triggers' : get_option('sandboxed_triggers'),
+    'Use seccomp' : libseccomp_dep.found(),
+    'Privileged group' : get_option('privileged_group'),
+    'Use dconf' : dconf_dep.found(),
+    'Use libsystemd' : libsystemd_dep.found(),
+    'Use libmalcontent' : malcontent_dep.found(),
+    'Use libzstd' : libzstd_dep.found(),
```

Comment:
Only in 0.57 can you pass these dep objects directly, rather than using .found(), but if you do then they will pretty-print the YES / NO values and also, if the dependency has a version, print the detected version. You may or may not care, just thought I'd note it.

Reply:
I don't really want to duplicate `summary()` into one version for (Meson >= 0.53 && Meson < 0.57) and one version for (Meson >= 0.57), so the question is: which is better, a `summary()` that works on versions >= 0.53 but has slightly less nice display than the state of the art, or a `summary()` that is only shown on versions >= 0.57 but looks as nice as possible? And I think this is a small enough enhancement that the answer is the more-backwards-compatible one.

Reply:
This would be a good situation to use "TODO" comments IMHO.